src/main/java/com/example/demo/userService/UserService.java

```diff
@@ -1,13 +1,54 @@
 package com.example.demo.userService;
 
+import com.example.demo.details.CustomUserDetails;
 import com.example.demo.entity.User;
+import com.example.demo.exceptions.UserNotFoundException;
+import com.example.demo.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
 
 import java.util.List;
+import java.util.Optional;
 
-public interface UserService {
+@Service
+public class UserService implements UserDetailsService {
```

Comment:
Usunalem interfejs, nie ma klasy UserServiceImpl i nie ma dziwnie wygladajacego CustomUserDetailsDervice.
W ogole dawanie nazw Custom i Implemention jest nienaturalne i tylko pokazuje, ze masz zbedna abstrakcje wprowadzona. Skoro nie masz dobrej nazwy na cos to raczej znaczy ze to cos nie powinno istniec